CrowsZA/functions/fn_surrenderChance.sqf

```diff
@@ -0,0 +1,137 @@
+	["_hesitation", 0, [0]]
+];
+
+
+if(isNull _unit) exitWith { false };
+
+
+// TODO: Add check if unit has already had this module applied/is already surrendered/etc
+
+
+if(_holdFire) then {
+	_unit setUnitCombatMode "BLUE";
+	_unit setCombatBehaviour "CARELESS";
+};
+
+
+private _trigger = createTrigger ["EmptyDetector", getPos _unit, false];
+_trigger setTriggerArea [_confrontationRadius, _confrontationRadius, 0, false];
+_trigger setTriggerActivation ["ANYPLAYER", "PRESENT", true];
+
+_trigger setVariable ["_unit", _unit];
+_unit setVariable ["_trigger", _trigger];
+_unit setVariable ["_hesitation", _hesitation];
+_unit setVariable ["_surrenderChance", _surrenderChance];
+
+_trigger attachTo [_unit];
+
+
+// TODO: Optionally ensure that the player pointing a weapon is considered HOSTILE to the units side
+// (don't mandate - could be useful to encourage friendly units, e.g. CIVs, to surrender too)
+
+private _condition = "
+if(!this) exitWith { false };
+private _unit = thisTrigger getVariable ""_unit"";
+{
+	[_unit, {
+		missionNamespace setVariable [
+			""LND_unit_callback"",
+			(toUpperANSI cameraView == ""GUNNER"") && (cursorObject == _this),
+			remoteExecutedOwner
+		];
+	}] remoteExecCall [""call"", _x];
+
+	if((missionNamespace getVariable ""LND_unit_callback"")) exitWith {true};
+} forEach thisList;
+(missionNamespace getVariable ""LND_unit_callback"")
+";
+
+
+private _activation = "
+private _unit = thisTrigger getVariable ""_unit"";
+private _surrenderChance = _unit getVariable ""_surrenderChance"";
+private _hesitation = _unit getVariable ""_hesitation"";
+
+deletevehicle thisTrigger;
+
+[_unit, _surrenderChance, _hesitation, thisTrigger] spawn {
+
+	params [""_unit"", ""_surrenderChance"", ""_hesitation""];
+
+	sleep ([random (0.5 + _hesitation), 0.5, _hesitation] call BIS_fnc_clamp);
+
+
+	if(random 1 < _surrenderChance) then {
+
+		if (crowsZA_common_aceModLoaded) then {
+			[""ace_captives_setSurrendered"",[_unit,true]] call CBA_fnc_globalEvent;
+		} else {
+			_unit action [""surrender"", _unit];
+		};
+		
+		_weapon = currentWeapon _unit;
+		_unit removeWeapon (currentWeapon _unit); 
+		_weaponHolder = ""WeaponHolderSimulated"" createVehicle [0,0,0]; 
+		_weaponHolder addWeaponCargoGlobal [_weapon,1]; 
+		_weaponHolder setPos (_unit modelToWorld [0,.2,1.2]); 
+		_weaponHolder disableCollisionWith _unit; 
+		_dir = random(360); 
+		_speed = 1; 
+		_weaponHolder setVelocity [_speed * sin(_dir), _speed * cos(_dir),4];
+	} else {
+		_unit setUnitCombatMode ""RED"";
+		_unit setCombatBehaviour ""COMBAT"";
+	};
+};
+";
+
+_trigger setTriggerStatements [
+	_condition,
+	_activation,
+	""
+];
+
+_unit addEventHandler ["Deleted", {
+	params ["_entity"];
+	deleteVehicle (_entity getVariable "_trigger");
+}];
+
+_unit addEventHandler ["Killed", {
+	params ["_unit", "_killer", "_instigator", "_useEffects"];
+	deleteVehicle (_unit getVariable "_trigger");
+}];
+
+
+_unit addEventHandler ["HandleDamage", {
```

Comment:
You could consider if the surrender-chance should increase if taking damage under a threshold. To simulate using non-lethal weapons, or limb shots. But if they do too much damage you go full reaction and fight.